angularSTEP/src/app/upload-recipe/upload-recipe.component.ts

```diff
@@ -0,0 +1,106 @@
+import { Component } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { AngularFirestore } from '@angular/fire/firestore';
+import { Router } from '@angular/router';
+import { formattedError } from '@angular/compiler';
+
+@Component({
+  selector: 'app-upload-recipe',
+  templateUrl: './upload-recipe.component.html',
+  styleUrls: ['./upload-recipe.component.scss'],
+})
+
+export class UploadRecipeComponent {
+  constructor(private db: AngularFirestore,
+              private router: Router,) {
+  }
+
+  ingredients: string[] = [];
+  instructions: string[] = [];
+  tools: string[] = [];
+
+  recipeForm = new FormGroup({
+    name: new FormControl('', [Validators.required]),
+    description: new FormControl('', [Validators.required]),
```

Comment:
This is where you could add another Validator from the mentioned suggestion up above.